### DISCUSSION TITLE
Loading local plugins

### DISCUSSION THREAD

**Discussion by WaelNalouti** [LazyVim/LazyVim#2202] [Q&A] [answered]
I am trying to build my first nvim plugin --so probably this is a newbie question lol.
where should the plugin config be so it can required by lazy automatically (when developing this plugin) without changing the rtp?
i have tried .config but it didn't work for some reason.

**Comment by dpetka2001** [comment]
Check out #1129. You basically need to specify `lazy.nvim`'s `dev.path` (see default [configuration](https://github.com/folke/lazy.nvim#%EF%B8%8F-configuration)).

**Comment by WaelNalouti** [comment]
thank for your help! 
I have updated `lazy.lua` to include `dev = {"myplugin"}` and cloned the repo in `~/projects` as described but i still get this error 
```
Error detected while processing ~/.config/nvim/init.lua:
E5113: Error while calling lua chunk: .../.local/share/nvim/lazy/lazy.nvim/lua/lazy/core/util.lua:56: attempt to index local 'path' (a nil value)
stack traceback:
        .../.local/share/nvim/lazy/lazy.nvim/lua/lazy/core/util.lua:56: in function 'norm'
        ...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/config.lua:216: in function 'setup'
        ...e/joe/.local/share/nvim/lazy/lazy.nvim/lua/lazy/init.lua:96: in function 'setup'
        ~/.config/nvim/lua/config/lazy.lua:9: in main chunk
        [C]: in function 'require'
        ~/.config/nvim/init.lua:2: in main chunk
```

**Reply by dpetka2001** [reply]
You need something like [this](https://github.com/dpetka2001/dotfiles/blob/5a73f6dcb1e02c70597942db4583a7b4c3139547/dot_config/nvim/lua/config/lazy.lua#L65-L68). In the `dev` section you must specify the `path` or leave it at default value which is `~/projects` and then you can either specify `patterns` with some pattern you want or specify `dev = true` inside the spec of a plugin. But this is for plugins that you fork the repo under `~/projects`.
~~I think what you want is `dir` in your case if you're developing a plugin locally and there's still no Github repo for your plugin.
So, create a file `~/.config/nvim/lua/plugins/myplugin.lua` with contents~~
```lua
return {
  { dir = "/local/path/to/myplugin" },
}
```
Not entirely sure, since I've never tried this. I've only forked repos locally and made `lazy.nvim` see the corresponding plugin as local. 

Edit: I read again that you said you cloned the repo, so don't do the `dir` part.

**Reply by WaelNalouti** [reply, marked as answer]
Thank you for the provided help i really appreciate it.
After some tries i get it to work by creating `~/.config/nvim/lua/pluginsmyplugin.lua` containing:
```lua
return {
    "myplugin",
    name = "myplugin",
    dev = {true}
}
```
Note that i have cloned the repo in `~/projects/` and i did not change `lazy.lua`.

**Reply by justrajdeep** [reply]
Hi @WaelNalouti 

i am stuck with the same issue. So you have cloned the repo locally under `~/projects` did you add the `dir` loading in your
 `init.lua` ? else how is it picking up your plugin from `~/projects` area? 
 
 or is your whole plugin now here `~/.config/nvim/lua/pluginsmyplugin.lua` ?

**Reply by WaelNalouti** [reply]
Hi @justrajdeep 

You need to clone the repo under `~/projects` then you create `~./config/nvim/lua/[plugin_name].lua` containing:

```lua
return {
    "[plugin_name]",
    name = "[plugin_name]",
    dev = {true}
}
```
don't forget to update `[plugin_name]` with your plugin name

**Reply by justrajdeep** [reply]
> Hi @justrajdeep
> 
> You need to clone the repo under `~/projects` then you create `~./config/nvim/lua/[plugin_name].lua` containing:
> 
> ```lua
> return {
>     "[plugin_name]",
>     name = "[plugin_name]",
>     dev = {true}
> }
> ```
> 
> don't forget to update `[plugin_name]` with your plugin name

I am confused about how neovim figures out where i have cloned the repo. Does it always have to be `~/projects` ?

**Reply by Old-Farmer** [reply]
> > Hi @justrajdeep
> > 
> > You need to clone the repo under `~/projects` then you create `~./config/nvim/lua/[plugin_name].lua` containing:
> > 
> > ```lua
> > return {
> >     "[plugin_name]",
> >     name = "[plugin_name]",
> >     dev = {true}
> > }
> > ```
> > 
> > don't forget to update `[plugin_name]` with your plugin name
> 
> I am confused about how neovim figures out where i have cloned the repo. Does it always have to be `~/projects` ?

All plugins is managed by lazy.nvim. Please check the doc of lazy.nvim.

**Reply by justrajdeep** [reply]
Thanks ... did not realize it 

After re-reading it, it is clear

```lua
      dev = {
        ---@type string | fun(plugin: LazyPlugin): string directory where you store your local plugin projects
        path = "~/projects",
        ---@type string[] plugins that match these patterns will use your local versions instead of being fetched from GitHub
        patterns = {}, -- For example {"folke"}
        fallback = false, -- Fallback to git when local plugin doesn't exist
      },
```
